src/Button.tsx

```diff
@@ -0,0 +1,192 @@
+   */
+  variant?: "normal" | "simple";
+}
+
+/**
+ * Style system for Space Kit buttons
+ *
+ * This is intended to be used as an abstraction for your project's style guide.
+ * There are way too many configuration options to use this component directly.
+ *
+ * @see https://zpl.io/amdN6Pr
+ */
+export const Button: React.FC<Props> = ({
+  children,
+  disabled = false,
+  fab = false,
+  icon,
+  size = "default",
+  variant = "normal",
+  ...otherProps
+}) => {
+  /**
+   * Icon size in pixels
+   *
+   * This is stored so we can use the same value for `height` and `width`
+   */
+  const iconSize = size === "small" ? 12 : size === "large" ? 24 : 16;
+
+  const iconOnly = !children;
+
+  if (fab) {
+    if (!icon) {
+      throw new TypeError("FAB buttons are required to have an `icon`");
+    } else if (children) {
+      throw new TypeError("FAB buttons cannot have children, only an `icon`");
+    }
+  }
+
+  return (
+    <button
+      {...otherProps}
+      disabled={disabled}
+      css={[
+        {
+          "&[disabled]": {
+            backgroundColor: colors.silver.dark,
+            color: colors.grey.light,
+
+            // We need to also set the `:hover` on `:disabled` so it has a higher
+            // specificity than any `:hover` classes passed in. This also means
+            // that both of these need to be overriden if we want to use a custom
+            // disabled color.
+            ":hover": {
+              backgroundColor: colors.silver.dark,
+              color: colors.grey.light
+            }
+          }
+        },
+
+        variant === "simple" && { backgroundColor: "transparent" },
+        {
+          borderRadius: fab ? "100%" : 4,
+
+          borderWidth: 0,
+          ...(variant === "normal" && {
+            boxShadow:
+              "0 1px 4px 0 rgba(18, 21, 26, 0.08), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)"
+          }),
+
+          minWidth: iconOnly
+            ? size === "small"
+              ? 28
+              : size === "large"
+              ? 42
+              : 36
+            : size === "small"
+            ? 76
+            : size === "large"
+            ? 112
+            : 100,
+
+          padding: size === "small" ? `5px 8px` : size === "large" ? 8 : 7,
+
+          ...(size === "small"
+            ? base.small
+            : size === "large"
+            ? base.large
+            : base.base),
+
+          fontWeight: 600,
+          // Disable the outline because we're setting a custom `:active` style
+          outline: 0,
+        },
+
+        // This must come after `:focus` or the `:focus` state will override `:active`
+        !disabled && {
+          ":hover, &[data-force-hover-state]": {
+            backgroundColor: colors.silver.base,
+            cursor: "pointer",
+            ...(variant === "normal" && {
+              // The `box-shadow` property is copied directly from Zeplin
+              boxShadow:
```

Comment:
I found the `...(condition && { ... })` syntax difficult to understand here. What if we used a ternary alternating `boxShadow` between a valid value and `none`?

```js
boxShadow:
  variant === 'normal'
    ? '0 5px 10px 0 rgba(18, 21, 26, 0.12), inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)' 
    : 'none'
```

Or better yet...

```js
const normalHoverShadow = [
  '0 5px 10px 0 rgba(18, 21, 26, 0.12)',
  'inset 0 0 0 1px rgba(18, 21, 26, 0.2)',
  'inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)'
];

boxShadow: variant === 'normal' ? normalHoverShadow.toString() : 'none'
```

Reply:
I think understanding what the `...()` means is important. That will either spread the values in the `{}` into the object or add nothing to the object.

Reply:
I also think this is adding more code and an unnecessary default back to the css default instead of doing the right thing: adding nothing in certain circumstances.

Reply:
For me, this is just about representing the conditional `boxShadow` style in a way that makes sense at first glance. If hardcoding `'none'` is not desirable, you could likely do this as well:

```js
boxShadow: variant === 'normal' && 'box shadow string'
```

If I remember correctly, Emotion will not include a box shadow property if the value is falsey.

Reply:
I understand the logic behind defining an array of the shadows and then using `toString()` on it; however that defeats the goal of copy-pasting the values in directly from zeplin. I opted to use the `? ... : 'none'` because I needed to to fix weird functionality glitches.

I'm going to leave this with the new update now 👍